vulnerabilities/data_dump.py

```diff
@@ -105,90 +97,96 @@ def archlinux_dump(extract_data):
     """
     Save data scraped from archlinux' security tracker.
     """
-    for item in extract_data:
-        cves = item['issues']
-        group = item['name']
-
-        advisories = set(item['advisories'])
-        vulnerabilities = cves + list(advisories)
-        vulnerabilities.append(group)
-        packages_name = item['packages']
-
-        affected_version = item['affected']
-        fixed_version = item['fixed']
-        if not fixed_version:
-            fixed_version = 'None'
-
-        vulnerability = Vulnerability.objects.create(
-            summary=item['type'],
-        )
+    base_url = 'https://security.archlinux.org'
 
-        for vulnerability_id in vulnerabilities:
-            VulnerabilityReference.objects.create(
-                vulnerability=vulnerability,
-                reference_id=vulnerability_id,
-                url=f'https://security.archlinux.org/{vulnerability_id}',
-            )
+    for avg in extract_data:
+        affected_packages = []
+        fixed_packages = []
 
-        for package_name in packages_name:
-            package_affected = Package.objects.create(
+        for package_name in avg['packages']:
+            ap, _ = Package.objects.get_or_create(
                 name=package_name,
                 type='pacman',
                 namespace='archlinux',
-                version=affected_version
-            )
-            ImpactedPackage.objects.create(
-                vulnerability=vulnerability,
-                package=package_affected
-            )
-            PackageReference.objects.create(
-                package=package_affected,
-                repository=f'https://security.archlinux.org/package/{package_name}',
+                version=avg['affected'],
```

Comment:
Is affected always a single version?

Reply:
It was a string in all the records I looked at so I'm pretty sure, but haven't tried to verify it any further.

In general I think it's not really a problem if the import crashes due to unexpected schemas. In fact, I'd argue we want it to crash as soon as possible instead of writing lots of empty values into the database due to overly defensive code.
We just need to make sure to be notified about any crashes that happen in cron jobs and such.